doc/specs/stdlib_logger.md

```diff
@@ -0,0 +1,965 @@
+  * `unformatted_in_error` - `unit` did not have a `form` specifier of
+    `'FORMATTED'`
+  * `unopened_in_error` - `unit` was not opened
+
+#### Example
+
+```fortran
+program demo_add_log_unit
+    use stdlib_logger, only: global_logger, read_only_error
+     ...
+     character(256) :: iomsg
+     integer :: iostat, unit, stat
+     ...
+     open( newunit=unit, 'error_log.txt',      &
+          form='formatted', status='replace', &
+          position='rewind', err=999,         &
+          action='read', iostat=iostat, iomsg=iomsg )
+     ...
+     call global_logger % add_log_unit( unit, stat )
+     select case ( stat )
+     ...
+     case ( read_only_error )
+         error stop 'Unable to write to "error_log.txt".'
+     ...
+     end select
+     ...
+     999 error stop 'Unable to open "error_log.txt".
+     ...
+end program demo_add_log_unit
+```
+
+### `configuration` - report a logger's configuration
+
+#### Status
+
+Experimental
+
+#### Description
+
+Reports the configuration of a logger.
+
+#### Syntax
+
+`call [[stdlib_logger(module):self % configuration(interface)]]( [ add_blankline, indent, max_width, time_stamp, log_units ] )`
+
+#### Class
+
+Pure subroutine
+
+#### Arguments
+
+`self`: shall be a scalar variable of type `logger_type`. It is an
+`intent(in)` argument. It shall be the logger whose configuration is reported.
+
+`add_blank_line` (optional): shall be a scalar default logical
+  variable. It is an `intent(out)` argument. A value of `.true.`
+  starts output with a blank line, and `.false.` otherwise.
+
+`indent` (optional): shall be a scalar default logical variable. It
+  is an `intent(out)` argument. A value of `.true.` indents subsequent
+  lines by four spaces, and `.false.` otherwise.
+
+`max_width` (optional): shall be a scalar default integer
+  variable. It is an `intent(out)` argument. A positive value bigger
+  than four defines the maximum width of the output, otherwise there
+  is no maximum width.
+
+`time_stamp` (optional): shall be a scalar default logical
+  variable. It is an `intent(out)` argument. A value of `.true.`
+  precedes output with a time stamp of the form 'yyyy-mm-dd
+  hh:mm:ss.sss', and `.false.` otherwise.
+
+`log_units` (optional): shall be a rank one allocatable array
+  variable of type default integer. It is an `intent(out)`
+  argument. On return it shall be the elements of the `self`'s `log_units`
+  array.
+
+#### Example
+
+```fortran
+    module example_mod
+      use stdlib_logger
+      ...
+      type(logger_type) :: logger
+    contains
+      ...
+      subroutine example_sub(unit, ...)
+        integer, intent(in) :: unit
+        ...
+        integer, allocatable :: log_units(:)
+        ...
+        call logger % configuration( log_units=log_units )
+        if ( size(log_units) == 0 ) then
+           call add_logger_unit( unit )
+        end if
+        ..
+      end subroutine example_sub
+      ...
+    end module example_mod
```

Comment:
```suggestion
module example_mod
    use stdlib_logger
   
    type(logger_type) :: logger
    contains
  
    subroutine example_sub(unit, ...)
        integer, intent(in) :: unit
  
        integer, allocatable :: log_units(:)
  
        call logger % configuration( log_units=log_units )
        if ( size(log_units) == 0 ) then
           call add_logger_unit( unit )
        end if
  
    end subroutine example_sub
  
end module example_mod
```

Reply:
Question: should we call this example of module `demo_example_mod`?

Reply:
I don't think prefixing with "demo" adds to the meaning. It's similar to "example" which is already in the name. I think it's fine as is, though I also wouldn't object to a more context-specific name.